internal/policy/filter_test.go

```diff
@@ -48,20 +49,53 @@ func TestRegexFilter(t *testing.T) {
 			extract:  `$1`,
 			expected: []string{"1", "2", "3"},
 		},
+		{
+			label: "valid pattern (complex regex)",
+			tags: []string{
+				"123-123.123.abcd123-debug",
+				"123-123.123.abcd123",
+				"123-123.123.abcd456-debug",
+				"123-123.123.abcd456",
+			},
+			pattern: `^(123-[0-9]+\.[0-9]+\.[a-z0-9]+-debug)`,
+			expected: []string{
+				"123-123.123.abcd123-debug",
+				"123-123.123.abcd456-debug",
+			},
+		},
+		{
+			label: "valid pattern with capture group (complex regex)",
+			tags: []string{
+				"123-123.123.abcd123-debug",
+				"123-123.123.abcd123",
+				"123-123.123.abcd456-debug",
+				"123-123.123.abcd456",
+			},
+			pattern: `^(?P<tag>123-[0-9]+\.[0-9]+\.[a-z0-9]+[^-debug])`,
+			extract: `$tag`,
+			expected: []string{
+				"123-123.123.abcd123",
+				"123-123.123.abcd456",
+			},
+		},
 	}
+
 	for _, tt := range cases {
 		t.Run(tt.label, func(t *testing.T) {
-			filter := newRegexFilter(tt.pattern, tt.extract)
-			filter.Apply(tt.tags)
-			r := sort.StringSlice(filter.Items())
-			if reflect.DeepEqual(r, tt.expected) {
-				t.Errorf("incorrect value returned, got '%s', expected '%s'", r, tt.expected)
+			f, err := NewRegexFilter(tt.pattern, tt.extract)
+			assert.NoError(t, err)
+
+			f.Apply(tt.tags)
+			r := sort.StringSlice(f.Items())
+			r.Sort()
+
+			if !assert.Equal(t, len(tt.expected), r.Len()) {
+				t.Fatalf("expected slice length to be %d, got %d", len(tt.expected), r.Len())
+			}
+
+			for i, expected := range tt.expected {
+				assert.Equal(t, expected, r[i])
```

Comment:
Hey, thanks for the suggestions. I will refactor this to use `gomega` like you suggested to maintain consistency across the codebase.

Reply:
https://github.com/fluxcd/image-reflector-controller/pull/458/commits/8e6987c660aa2a82229c3c4a99f7fc54ee0cfce0 Let me know if that aligns. 

Btw I was curious, why is `gomega` preferred over `testify`? I'd love to know the motivation behind it so that I can start tinkering with it if it's better.

Reply:
Makes sense. Thanks for the clarification.